src/com/github/Kraken3/AFKPGC/BotDetector.java

```diff
@@ -391,5 +354,14 @@ public static void parseBanlist() {
 			AFKPGC.logger.warning("Error while trying to parse the banned players file");
 		}
 	}
+	
+	public static void warnPlayer(Player p) {
+		if (p != null) {
+			p.sendMessage("[WARNING] You were determined to be a cause of lag, if you dont stay out of the area "+
+					"you are in right now, you will be temporarily banned");

```

Comment:
Should probably be something more specific like "You are in a region with a high concentration of lag sources. Please immediately depart the area (leave render distance) or you will be temporarily banned."